src/solvers/indicators.jl

```diff
@@ -0,0 +1,85 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    IndicatorHennemannGassnerShallowWater(equations::AbstractEquations, basis;
+                                          alpha_max=0.5,
+                                          alpha_min=0.001,
+                                          alpha_smooth=true,
+                                          variable)
+
+Modified version of the [`IndicatorHennemannGassner`](@ref)
+indicator used for shock-capturing for shallow water equations. After
+the element-wise values for the blending factors are computed an additional check
+is made to see if the element is partially wet. In this case, partially wet elements
+are set to use the pure finite volume scheme that is guaranteed to be well-balanced
+for this wet/dry transition state of the flow regime.
+
+See also [`VolumeIntegralShockCapturingHG`](@ref).
+
+## References
+
+- Hennemann, Gassner (2020)
+  "A provably entropy stable subcell shock capturing approach for high order split form DG"
+  [arXiv: 2008.12044](https://arxiv.org/abs/2008.12044)
+"""
+struct IndicatorHennemannGassnerShallowWater{RealT <: Real, Variable, Cache} <:
+       Trixi.AbstractIndicator
+    alpha_max::RealT
+    alpha_min::RealT
+    alpha_smooth::Bool
+    variable::Variable
+    cache::Cache
+end
+
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+# of the shallow water equations
+# It modifies the shock-capturing indicator to use full FV method in dry cells
```

Comment:
```suggestion
# It modifies the shock-capturing indicator to use full FV method in dry cells
# or partially dry cells containing a wet/dry transition.
```

Reply:
Yes that was not properly commented before. In addition I think this should be elements instead  of cells.
I applied these changes in 77f44c650f95e4bc7bcbd284ecd2f60f6b9d0609